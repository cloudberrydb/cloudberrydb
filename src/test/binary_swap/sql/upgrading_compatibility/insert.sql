INSERT INTO sales_heap (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;


INSERT INTO sales_ao (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;


INSERT INTO sales_aocs (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;

INSERT INTO sales_partition_heap (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;

INSERT INTO sales_partition_ao (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;

INSERT INTO sales_partition_aocs (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;

SELECT COUNT(*) FROM sales_heap;

SELECT COUNT(*) FROM sales_ao;

SELECT COUNT(*) FROM sales_aocs;

SELECT COUNT(*) FROM sales_partition_heap;

SELECT COUNT(*) FROM sales_partition_ao;

SELECT COUNT(*) FROM sales_partition_aocs;
