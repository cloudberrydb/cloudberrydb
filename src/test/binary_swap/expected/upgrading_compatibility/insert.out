INSERT INTO sales_heap (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
INSERT INTO sales_ao (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
INSERT INTO sales_aocs (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
INSERT INTO sales_partition_heap (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
INSERT INTO sales_partition_ao (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
INSERT INTO sales_partition_aocs (
  product_id,
  is_audited,
  quantity,
  total_sales,
  unit_price,
  discount,
  description,
  sale_date,
  order_date,
  status,
  customer_name,
  price
)
SELECT
  x.id, -- product_id
  CASE
    WHEN random() < 0.5 THEN TRUE
    ELSE FALSE
  END AS is_audited, -- is_audited
  (random() * 100)::smallint, -- quantity
  (random() * 1000000)::bigint, -- total_sales
  (random() * 100)::real, -- unit_price
  (random() * 1)::double precision, -- discount
  'Product description ' || x.id, -- description
  now() - INTERVAL '1 day' * floor(random() * 365 * 3), -- sale_date
  '2023-01-01'::DATE + INTERVAL '1 day' * floor(random() * 365 * 3), -- order_date
  CASE
    WHEN random() < 0.05 THEN 'Cancelled'
    WHEN random() < 0.5 THEN 'Closed'
    ELSE 'Processing'
  END AS status, -- status
  'Customer ' || x.id, -- customer_name
  (random() * 1000000)::decimal(20, 10) -- price
FROM (
  SELECT * FROM generate_series(1000001, 1010000) AS id
) AS x;
SELECT COUNT(*) FROM sales_heap;
 count 
-------
 20000
(1 row)

SELECT COUNT(*) FROM sales_ao;
 count 
-------
 20000
(1 row)

SELECT COUNT(*) FROM sales_aocs;
 count 
-------
 20000
(1 row)

SELECT COUNT(*) FROM sales_partition_heap;
NOTICE:  One or more columns in the following table(s) do not have statistics: sales_partition_heap
HINT:  For non-partitioned tables, run analyze <table_name>(<column_list>). For partitioned tables, run analyze rootpartition <table_name>(<column_list>). See log for columns missing statistics.
 count 
-------
 20000
(1 row)

SELECT COUNT(*) FROM sales_partition_ao;
NOTICE:  One or more columns in the following table(s) do not have statistics: sales_partition_ao
HINT:  For non-partitioned tables, run analyze <table_name>(<column_list>). For partitioned tables, run analyze rootpartition <table_name>(<column_list>). See log for columns missing statistics.
 count 
-------
 20000
(1 row)

SELECT COUNT(*) FROM sales_partition_aocs;
 count 
-------
 20000
(1 row)

